hack/update-helm-index.sh

```diff
@@ -0,0 +1,127 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+this=`basename $0`
+
+usage () {
+cat << EOF
+Usage: $this [-h] [--from-url URL] [--from-local PATH]
+
+Options:
+  --from-url URL        specify the URL to download the Helm chart(s) archive(s) from
+						all Helm chart archives with the .tgz extension will be downloaded
+						and added to the Helm repo
+  --from-local PATH     specify the local path to the Helm chart archive (tgz file)
+						only one chart archive can be specified
+  --help/-h             show this help and exit
+
+NOTE: only one of --from-url or --from-local can be specified
+
+Example:
+
+  $this --from-url https://github.com/NVIDIA/k8s-device-plugin/archive/refs/tags/nvidia-device-plugin-0.15.0.tgz
+
+EOF
+}
+
+#
+# Parse command line
+#
+asset_url=
+asset_local=
+while [[ $# -gt 0 ]]; do
+	key="$1"
+	case $key in
+		--from-url)
+			asset_url="$2"
+			shift 2
+			break
+			;;
+		--from-local)
+			asset_local="$2"
+			shift 2
+			break
+			;;
+		--help/h) usage
+			exit 0
+			;;
+		*) usage
+			exit 1
+			;;
+	esac
+done
+
+GH_REPO_PULL_URL="https://github.com/NVIDIA/k8s-device-plugin.git"
+git clone --local $$(pwd) $(HELM_REPO_PATH)
+git -C $(HELM_REPO_PATH) remote add github $(GH_REPO_PULL_URL)
+git -C $(HELM_REPO_PATH) fetch github gh-pages
+git -C $(HELM_REPO_PATH) checkout gh-pages
+
+pushd $HELM_REPO_PATH/stable > /dev/null
+# Download charts from release assets
+
+if [ -n "$asset_local" ]; then
+	echo "Copying $asset_local..."
+	cp -f $asset_local $HELM_REPO_PATH/stable
+fi
+
+if [ -n "$asset_url" ]; then
+	for asset_url in $asset_urls; do
+	    if ! echo "$asset_url" | grep -q '.*tgz$'; then
+	        echo "Skipping $asset_url, does not look like a Helm chart archive"
+	        continue
+	    fi
+
+	    echo "Downloading $asset_url..."
+	    curl -sSfLO $asset_url
+	    # We rely on all release assets having the same baseurl
+	    download_baseurl=`dirname $asset_url`
+	done
+
+	if [ -z "$download_baseurl" ]; then
+	    echo "No Helm chart release assets found"
+	    exit 0
+	fi
+fi
+
+echo "Updating helm index"
+helm repo index stable --merge index.yaml --url https://nvidia.github.io/k8s-device-plugin/stable
+popd > /dev/null
+
+cp -f $HELM_REPO_PATH/stable/index.yaml $HELM_REPO_PATH/index.yaml
+
+pushd $HELM_REPO_PATH > /dev/null
+# Check if there were any changes in the repo
+if [ -z $(git status --short) ]; then
+    echo "No changes in Helm repo index, gh-pages branch already up-to-date"
+    exit 0
+fi
+
+# Create a new commit
+commit_msg="Update helm charts for release $VERSION
+
+This commit was created by the k8s-device-plugin release workflow.
+It updates the Helm charts in the gh-pages branch to include the $VERSION release assets.
+Also updates the Helm index.yaml file to include the new chart versions."
+
+echo "Committing changes..."
+git add index.yaml stable
+git commit -m "$commit_msg"
```

Comment:
Do we need to add `-s`?

Reply:
no... use of habit will remove

Reply:
I mean do we want to sign-off the commits, not sign the commits.

Reply:
they will be signed off with `no-reply@github.com`, done